cli/cloe_launch/__init__.py

```diff
@@ -63,76 +57,24 @@ def __init__(self, profile: str = None):
                 self._conf[k] = conf[k]
-            self.default_profile = self._conf["default_profile"]
-
-        # Read all profile names from profile directory
-        self.all_profiles = [
-            f
-            for f in os.listdir(self.profiles_dir)
-            if os.path.isfile(self.profile_path(f))
-        ]
-
-        # Set current profile
-        if profile is not None:
-            self.set_current(profile)
-        elif self.default_profile is not None:
-            self.set_current(self.default_profile)
-
-    def profile_path(self, profile: str) -> str:
-        """Return the path to named profile."""
-        return os.path.join(self.profiles_dir, profile)
 
-    def profile_runtime(self, profile: str) -> str:
+    def profile_runtime(self, hash: str) -> str:
         """Return the path to the runtime directory of the profile."""
-        return os.path.join(self.runtime_dir, profile)
+        return os.path.join(self.runtime_dir, hash)
 
-    def set_current(self, profile: str) -> None:
-        """Set the current profile and make sure it exists."""
-        self.current_profile = profile
-
-    def set_default(self, profile: str) -> None:
-        """Set the default profile and write it to the configuration."""
-        if profile is not None and profile not in self.all_profiles:
-            raise ConfigurationError(f"profile {profile} does not exist")
-        self.default_profile = profile
-        self._conf["default_profile"] = profile
+    def write(self) -> None:
+        """Write current configuration to the disk."""
         logging.info(f"Write configuration to {self.config_file}:\n  {self._conf}")
-        with open(self.config_file, "w") as file:
+        with open(self.config_file, "w", encoding="utf-8") as file:
             toml.dump(self._conf, file)
 
-    def read(self, profile: str) -> str:
-        """Read the specified profile."""
-        logging.info("Open:", self.profile_path(profile))
-        with open(self.profile_path(profile)) as file:
-            return file.read()
-
-    def edit(self, profile: str, create: bool = False) -> None:
-        """Open the specified profile in the user's editor."""
+    def edit(self, create: bool = False) -> None:
+        """Open the configuration in the user's editor."""
         editor = os.getenv("EDITOR")
         if editor is None:
             raise ConfigurationError("environment variable EDITOR is unset")
-        if not create and not os.path.exists(self.profile_path(profile)):
-            raise ConfigurationError(f"profile {profile} does not exist")
-        cmd = [editor, self.profile_path(profile)]
-        logging.info("Exec:", " ".join(cmd))
+        if not create and not os.path.exists(self.config_file):
+            raise ConfigurationError(f"configuration {self.config_file} does not exist")
+        cmd = [editor, self.config_file]
+        logging.info("Exec: %s", " ".join(cmd))
         subprocess.call(cmd)
```

Comment:
Since we don't capture output, the user will see if their editor has a problem.
Otherwise, a proper error is raised, and in the case that EDITOR is set incorrectly, an error will also be raised. This super-rare edge case is not worth writing fancy error messages for, imo.

Reply:
However, I have added a sub-command to even make use of this method, since it was missing ^^